volatility3/framework/symbols/windows/extensions/__init__.py

```diff
@@ -633,7 +644,11 @@ def is_valid(self) -> bool:
             if dtb & ~0xFFF == 0:
                 return False
 
-            ## TODO: we can also add the thread Flink and Blink tests if necessary
+            # Quick smear test on thread Flink and Blink
+            kernel = 0x80000000  # Yes, it's a quick test
```

Comment:
This... doesn't feel robust, and it also feels like it only works for 32-bit systems?  I mean, it'll work to filter out *some* bad entries, but?  Is there anything we can do about this or does it really need to be this way?

Reply:
Yeah, I thought about this before submitting the PR. The idea of this is to do a quick check, it doesn't need to be perfect. Also, that will be the safer way to go supporting both 32/64bit as it only filter out the lowest addresses. 
```python
           if list_head.Flink < kernel or list_head.Blink < kernel:
                return False
```
In 64bit systems, there will be cases where the link_head addresses fall within the user virtual address space or in the canonical hole (invalid), and this filter won't catch them. We could use `0xffff800000000000` in those cases, however, we would likely face similar issues to those you mentioned [here](https://github.com/volatilityfoundation/volatility3/issues/1162#issuecomment-2209502111)

Reply:
Ok, I'd sooner get some bad apples than weed out valid ones, so happy to stick with this.

As per another conversation though, I think that should be 0xffff000000000000 rather than 0xffff800000000000 unless we're intentionally trying to keep out kernel memory.